### ISSUE TITLE
connections not limited to node specified in --add-exclusive-node

### ISSUE THREAD

**Issue by radfish** [monero-project/bitmonero#848]
The daemon connects to other nodes besides the exclusive one.

Also, tried with --hide-my-port: same behavior.
Also, tried rm p2pstate.bin, same behavior.

Expected behavior: there is only one connection made from or to bitmonerod when --add-exclusive-node is specified.

**Comment by osensei** [comment]
Currently, what --add-exclusive-node does is instruct the daemon to only make outgoing connections to the exclusive nodes. That doesn't prevent other peers from connecting to your node.

What --hide-my-port does is tell the daemon to not announce itself to other nodes as a peer candidate.

BUT.... if your node has previously announced itself to the network you launch it with "--hide-my-port --add-exclusive-daemon xxx.xxx.xxx.xxx:18080" other peers will already know about your node and they may still try to connect to it (and succeed).

In this case, to accomplish the behavior you are expecting, you should run your node in a port that you haven't used before on your current IP, that is... using the parameter --p2p-bind-port. Or, if you are behind a NAT router, using --no-igd will prevent your router from automatically forwarding port 18080 to your node and that would be enough to prevent other nodes from connecting to you (as long as that port is not already forwarded on your router).

In summary, if other nodes know about the existence of your node, they may still connect to you despite using --add-exclusive-node, so what you need to do is to ensure no other node will reach yours.

**Comment by radfish** [comment]
Thanks for the detailed explanation. This all makes sense.

**Comment by iamsmooth** [comment]
This is an old issue which I just noticed but a better workaround than a new port (which could still in theory accept a connection if someone found it and firewall allowed) is `--p2p-bind-ip 127.0.0.1`. That makes it impossible for any outside connections since the p2p listener is bound to the loopback.
